core/src/Cabana_Sort.hpp

```diff
@@ -589,6 +589,75 @@ void permute( const BinningDataType& binning_data,
     Kokkos::fence();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Given binning data permute a slice.
+
+  \tparam BinningDataType The binning data type.
+
+  \tparm SliceType The slice type.
+
+  \param binning_data The binning data.
+
+  \param slice The slice to permute.
+ */
+template<class BinningDataType, class SliceType>
+void permute( const BinningDataType& binning_data,
+              SliceType& slice,
+              typename std::enable_if<(is_binning_data<BinningDataType>::value &&
```

Comment:
@dalg we need the `is_slice` for the compiler to pick the correct implementation. If we move `is_binning_data` to a `static_assert` will compile times improve?

Reply:
talked with @dalg24 on this - we will resolve elsewhere